util/dvsim/SlurmLauncher.py

```diff
@@ -46,28 +47,30 @@ def _do_launch(self):
 
         self._dump_env_vars(exports)
 
+        # Add a command delimiter if necessary
+        slurm_setup_cmd = SLURM_SETUP_CMD
+        if slurm_setup_cmd != '' and not slurm_setup_cmd.endswith(';'):
+            slurm_setup_cmd += ';'
+
+        # Encapsulate the run command with the slurm invocation
+        slurm_cmd = f'srun -p {SLURM_QUEUE} --mem={SLURM_MEM} --mincpus={SLURM_MINCPUS} ' \
+                    f'--time={SLURM_TIMEOUT} --cpus-per-task={SLURM_CPUS_PER_TASK} ' \
+                    f'bash -c "{slurm_setup_cmd} {self.deploy.cmd}"'
+
         try:
             with open(self.slurm_log_file, 'w') as out_file:
                 out_file.write("[Executing]:\n{}\n\n".format(self.deploy.cmd))
                 out_file.flush()
 
-                # Add a command delimiter if necessary
-                slurm_setup_cmd = SLURM_SETUP_CMD
-                if slurm_setup_cmd != '' and not slurm_setup_cmd.endswith(';'):
-                    slurm_setup_cmd += ';'
-
-                # Encapsulate the run command with the slurm invocation
-                slurm_cmd = f'srun -p {SLURM_QUEUE} --mem={SLURM_MEM} --mincpus={SLURM_MINCPUS} ' \
-                            f'--time={SLURM_TIMEOUT} --cpus-per-task={SLURM_CPUS_PER_TASK} ' \
-                            f'bash -c "{slurm_setup_cmd} {self.deploy.cmd}"'
-
                 log.info(f'Executing slurm command: {slurm_cmd}')
                 self.process = subprocess.Popen(shlex.split(slurm_cmd),
                                                 bufsize=4096,
                                                 universal_newlines=True,
                                                 stdout=out_file,
                                                 stderr=out_file,
                                                 env=exports)
+        except IOError as e:
+            raise LauncherError(f'File Error: {e} when handling {self.slurm_log_file}')
```

Comment:
Nit: Maybe worth adding a '\n' here like the next error? I don't have experience of this error type but I *think* it's the same as OSError and you get more than a single word of text?